[GraphQL] Add some documentation about queries and mutations (#761)
api-platform/docs: core/graphql.md

# ...            
```

## Queries

If you don't know what a query is yet, the document about them is [here](https://graphql.org/learn/queries/).

For each resource, two queries are available: one for retrieving an item and the other one for the collection.
For example, if you have a `Book` resource, the queries `book` and `books` can be used.

### Global Object Identifier

When querying an item, you need to pass an identifier as argument. Following the [Relay Global Object Identification Specification](https://facebook.github.io/relay/graphql/objectidentification.htm),

> the identifier needs to be globally unique. That's why in API Platform, you have to use the **IRI** of the resource for this argument.

Reviewer: ```suggestion
the identifier needs to be globally unique. In API Platform, this argument will be represented as an [IRI (Internationalized Resource Identifier)](https://www.w3.org/TR/ld-glossary/#internationalized-resource-identifier).
```

WDYT? I'd definitely add the link to the w3.org website.

> For example, to query a book having the id 89, you have to do:

Reviewer: ```suggestion
For example, to query a book having as identifier `89`, you have to query the following:
```

```graphql
{
  book(id: "/books/89") {
    title
    isbn
  }
}
```

> 

Reviewer: > Note that in this example, we're retrieving two fields: `title` and `isbn`.

### Pagination

API Platform natively enables a cursor-based pagination for collections.
It supports [GraphQL's Complete Connection Model](https://graphql.org/learn/pagination/#complete-connection-model) and is compatible with [Relay's Cursor Connections Specification](https://facebook.github.io/relay/graphql/connections.htm).

Here is an example query leveraging the pagination system:

```graphql
{
  offers(first: 10, after: "cursor") {
    totalCount
    pageInfo {
      endCursor
      hasNextPage
    }
    edges {
      cursor
      node {
        id
      }
    }
  }
}
```

A pair of parameters works with the query: `first` and `after`.

> More precisely:

Reviewer: The two parameters `first` and `after` are necessary to make the paginated query work, more precisely:

* `first` corresponds to the items per page starting from the beginning;
* `after` corresponds to the `cursor` from which the items are returned.

The current page always has an `endCursor` present in the `pageInfo` field.

> 

Reviewer: ```suggestion
```

To get the next page, you would add the `endCursor` from the current page as the `after` parameter.

```graphql
{
  offers(first: 10, after: "endCursor") {
  }
}
```

How do you know when you have reached the last page? It is the aim of the property `hasNextPage` in `pageInfo`.

> When it is false, you know it is the last page and moving forward will give you an empty result.

Reviewer: When the property `hasNextPage` of the `pageInfo` field is `false`, you've reach the last page. If you move forward, you'll end up having an empty result.

## Mutations

> If you don't know what a mutation is yet, the document about them is [here](https://graphql.org/learn/queries/#mutations).

Reviewer: a mutation => them ? `it` would be better?

> For each resource, three mutations are available: one for creating it (`create`), one for updating it (`update`) and one for deleting it (`delete`).

Reviewer: one it repetition could be avoided?

Author: I don't know how :confused:
